File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max = 0;
        int low = prices[0];
        for(int i = 1; i < prices.size(); i++){
            if(prices[i] - low > max){
                max = prices[i] - low;
            }else if(prices[i]< low){
                low = prices[i];
            }
        }
        return max;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        curr = 0
        lowest = prices[0]
        for i in range(len(prices)):
            lowest = min(lowest, prices[i])
            curr = max(curr, prices[i]-lowest)
        
        return curr

            